### PR TITLE
Oxygen XML Editor: use unversioned URL

### PR DESCRIPTION
**Pull request by adunning**
Builds can change without a version update; see #28632 and #115739.

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [X] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [X] `brew audit --cask <cask>` is error-free.
- [X] `brew style --fix <cask>` reports no offenses.

### CODE REVIEW

File: Casks/oxygen-xml-editor.rb [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
 
   url "https://www.oxygenxml.com/InstData/Editor/MacOSX/VM/oxygen#{arch}.dmg"
   name "oXygen XML Editor"
   desc "Tools for XML editing, including Oxygen XML Developer and Author"
   homepage "https://www.oxygenxml.com/xml_editor.html"
 
+  livecheck do
+    url "https://www.oxygenxml.com/allVersions.xml"
+    strategy :page_match do |page|
+      v = page[/buildId_(\d+(?:_\d+)+)/i, 1]
```
Comment (neutric) [inline Casks/oxygen-xml-editor.rb]: Not sure if this edit is a reaction to my remark, but if so, I was talking about `Build id: 2021121317` as seen [here](https://www.oxygenxml.com/xml_editor/download_oxygenxml_editor.html) or [here](https://www.oxygenxml.com/build_history.html).

**Comment by miccal** [comment]
@adunning can you please elaborate on which one of the `oxygen.dmg` and the `oxygen-openjdk.dmg` versions are recommended for Intel?

**Comment by joewiz** [comment]
@adunning Thank you so much! 

@miccal Indeed, the default version offered on the oXygen download page for both Intel and Apple Silicon Macs is now the OpenJDK 17 version. For years and up until yesterday, the default for Intel Macs was the Java SE 8 version. I inquired with the oXygen developers, and they explained that they were working toward making the OpenJDK 17 version the default. They just released a new build of oXygen 24.0 addressing some issues with OpenJDK 17 on Mac. So I think we would now be in good shape setting the OpenJDK 17 edition as the default for both Intel and Apple Silicon Macs.

The updated livecheck directive looks like it is now sensitive to the oXygen releases of updated builds, as published at https://www.oxygenxml.com/rssBuildID.xml. 

I wonder if the new "build sensitivity" added in this PR might also allow us to actually adopt the sha256 directive. The hashes for each binary are published, so I think they are stable enough for us to rely on. For example, you will see them linked at https://www.oxygenxml.com/xml_editor/download_oxygenxml_editor.html?os=macOS, as follows, for each build:

- Intel OpenJDK 17: https://www.oxygenxml.com/InstData/Editor/MacOSX/VM/oxygen-openjdk.dmg_sha256sum.html
- Apple Silicon OpenJDK 17: https://www.oxygenxml.com/InstData/Editor/MacOSX/VM/oxygen-openjdk-aarch64.dmg_sha256sum.html
- Intel Java SE 8: https://www.oxygenxml.com/InstData/Editor/MacOSX/VM/oxygen.dmg_sha256sum.html

**Comment by joewiz** [comment]
p.s. Checking out the PR, I can confirm it correctly detects the new build and downloads it:

```
% brew upgrade
Running `brew update --preinstall`...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
No changes to formulae.

==> Casks with 'auto_updates true' or 'version :latest' will not be upgraded; pass `--greedy` to upgrade them.
==> Upgrading 1 outdated package:
oxygen-xml-editor 24.0 -> 24.0,2021121317
==> Upgrading oxygen-xml-editor
==> Downloading https://www.oxygenxml.com/InstData/Editor/MacOSX/VM/oxygen.dmg
######################################################################## 100.0%
Warning: No checksum defined for cask 'oxygen-xml-editor', skipping verification.
==> Backing App Suite 'Oxygen XML Editor' up to '/usr/local/Caskroom/oxygen-xml-editor/24.0/Oxygen XML Editor'
==> Removing App Suite '/Applications/Oxygen XML Editor'
==> Moving App Suite 'Oxygen XML Editor' to '/Applications/Oxygen XML Editor'
==> Purging files for version 24.0 of Cask oxygen-xml-editor
🍺  oxygen-xml-editor was successfully upgraded!
```

**Comment by miccal** [comment]
> @miccal Indeed, the default version offered on the oXygen download page for both Intel and Apple Silicon Macs is now the OpenJDK 17 version. For years and up until yesterday, the default for Intel Macs was the Java SE 8 version. I inquired with the oXygen developers, and they explained that they were working toward making the OpenJDK 17 version the default. They just released a new build of oXygen 24.0 addressing some issues with OpenJDK 17 on Mac. So I think we would now be in good shape setting the OpenJDK 17 edition as the default for both Intel and Apple Silicon Macs.

Thank you @joewiz, I have changed this to download the OpenJDK version for both Intel and Arm.

@adunning please feel free to submit the Java 8 version of this to the [`homebrew-cask-versions`](https://github.com/Homebrew/homebrew-cask-versions) repository if you like.

> I wonder if the new "build sensitivity" added in this PR might also allow us to actually adopt the sha256 directive. The hashes for each binary are published, so I think they are stable enough for us to rely on. 

As the `url`'s are un-versioned these should stay as they are with `sha256 :no_check` -- only the older versions have published versioned links -- https://www.oxygenxml.com/xml_editor/software_archive_editor.html

**Comment by joewiz** [comment]
@miccal Makes sense! Thank you!

**Pull request merged**